lib/backup/database/postgresql.rb

```diff
@@ -22,6 +22,10 @@ class Error < Backup::Error; end
       attr_accessor :host, :port, :socket
 
       ##
+      # URL includes credentials and connectivity options
+      attr_accessor :url
```

Comment:
we should probably also output a warning when this option is used in combination with the username, password, name, host, port, socket, etc options, otherwise people might use both while most options are ignored when the `url` option is used.
Should include some testing for that as well

For example:

```ruby
def check_options
  return unless url
  ignored_options = {
    username: username,
    password: password,
    host: host,
    port: port,
    socket: socket,
    name: name
  }.reject { |_, value| value.nil? }
  return if ignored_options.empty?

  Logger.warn "PostgreSQL: the options " \
    "#{ignored_options.keys.join(", ")} are set, but will be " \
    "ignored because the `:url` option is set."
end
```

and call this method in the perform method, or in the initializer.